core/src/main/scala/org/apache/spark/scheduler/cluster/CoarseGrainedSchedulerBackend.scala

```diff
@@ -631,8 +635,10 @@ class CoarseGrainedSchedulerBackend(scheduler: TaskSchedulerImpl, val rpcEnv: Rp
       force: Boolean): Seq[String] = {
     logInfo(s"Requesting to kill executor(s) ${executorIds.mkString(", ")}")
 
+    val idleExecutorIds = executorIds.filter { id => force || !scheduler.isExecutorBusy(id) }
```

Comment:
Nit: I would not use the name `idleExecutorIds` for this variable as when flag `force` is true then not only idle executors are contained.

Reply:
meh, I'm not sure what else you'd call it ... there is already `executorsToKill` lower down ... unless you have a better suggestion, `idleExecutorIds` is probably good enough

but this does leave a small race for SPARK-19757, doesn't it?  After this executes, then an executor gets a task scheduled on it so its no longer idle, but you still kill it below?  To really prevent that, you'd need to get both locks (in the same order of course) so

```
val response = scheduler.synchronized { this.synchronized {
```

it also isn't the worst thing in the world if we occasionally kill an executor which just got a task scheduled on it.

Reply:
@squito  If you wanted to prevent that then you need to put the `val idleExecutorIds = executorIds.filter { id => force || !scheduler.isExecutorBusy(id) }` back inside that scheduler lock, right?

> it also isn't the worst thing in the world if we occasionally kill an executor which just got a task scheduled on it.

So we don't count this as a task failure right? Not sure where to look to verify that.

Reply:
yes, I meant with the filter happening inside both locks -- more like it was before the current form of the PR, or as you suggested

Reply:
I have a suggestion for naming but I do not insist on that:
- renaming  `idleExecutorIds ` to `executorsToKill`
- renaming the old `executorsToKill` to `knownExecutorsToKill`

I also have checked the `synchronised` blocks of `CoarseGrainedSchedulerBackend` and its derived classes and have not found any other place where the scheduler is used for locking (within the synchronised block).